<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
 <HEAD>
   <TITLE> [Plantlore-dev] plantlore chyby
   </TITLE>
   <LINK REL="Index" HREF="http://lists.berlios.de/pipermail/plantlore-dev/2007-January/index.html" >
   <LINK REL="made" HREF="mailto:plantlore-dev%40lists.berlios.de?Subject=Re%3A%20%5BPlantlore-dev%5D%20plantlore%20chyby&In-Reply-To=%3COF7DBCEF09.DE8C14F1-ONC1257258.0027FB2D-C1257258.002892C4%40schkocr.cz%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <style type="text/css">
       pre {
           white-space: pre-wrap;       /* css-2.1, curent FF, Opera, Safari */
           }
   </style>
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="001336.html">
   <LINK REL="Next"  HREF="001338.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[Plantlore-dev] plantlore chyby</H1>
    <B>Vladimir Hans</B> 
    <A HREF="mailto:plantlore-dev%40lists.berlios.de?Subject=Re%3A%20%5BPlantlore-dev%5D%20plantlore%20chyby&In-Reply-To=%3COF7DBCEF09.DE8C14F1-ONC1257258.0027FB2D-C1257258.002892C4%40schkocr.cz%3E"
       TITLE="[Plantlore-dev] plantlore chyby">vladimir.hans at nature.cz
       </A><BR>
    <I>Wed Jan  3 08:23:06 CET 2007</I>
    <P><UL>
        <LI>Previous message: <A HREF="001336.html">[Plantlore-dev] Poznamky k pripominkam a chybam
</A></li>
        <LI>Next message: <A HREF="001338.html">[Plantlore-dev] Pracovni schuzka
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#1337">[ date ]</a>
              <a href="thread.html#1337">[ thread ]</a>
              <a href="subject.html#1337">[ subject ]</a>
              <a href="author.html#1337">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>Ahoj,
pos&#237;l&#225;m dal&#353;&#237; chybky. P&#345;&#225;n&#237;&#269;ka po&#353;lu, a&#382; to proberu s ostatn&#237;mi a 
zakomponuji do toho v&#283;ci z Botanick&#233;ho &#250;stavu AV&#268;R.

Nesn&#225;&#353;&#237; to konkuren&#269;n&#237; p&#345;&#237;stup k dat&#367;m. Jestli&#382;e m&#225;m spu&#353;t&#283;n&#233; 2 instance 
plantlore (ob&#283; spu&#353;t&#283;n&#233; z netbeans) a na obou otev&#345;en&#253; form pro p&#345;id&#225;v&#225;n&#237; 
n&#225;lez&#367;, tak po &#269;ase vznikne z&#225;mek na tabulce thabitats a operaci nelze 
prov&#233;st. 
V netbeans to hodilo hl&#225;&#353;ku (instance, kter&#225; odm&#237;tla prov&#233;st operaci):
net.sf.plantlore.common.exception.DBLayerException: P&#345;ipojen&#237; se nezda&#345;ilo
        at 
net.sf.plantlore.server.HibernateDBLayer.initialize(HibernateDBLayer.java:238)
        at 
net.sf.plantlore.client.login.Login$ConnectionTask.task(Login.java:336)
        at net.sf.plantlore.common.Task$1.construct(Task.java:200)
        at net.sf.plantlore.common.SwingWorker$2.run(SwingWorker.java:108)
        at java.lang.Thread.run(Thread.java:595)
Caused by: org.hibernate.exception.JDBCConnectionException: Cannot open 
connection
        at 
org.hibernate.exception.SQLStateConverter.convert(SQLStateConverter.java:74)
        at 
org.hibernate.exception.JDBCExceptionHelper.convert(JDBCExceptionHelper.java:43)
        at 
org.hibernate.exception.JDBCExceptionHelper.convert(JDBCExceptionHelper.java:29)
        at 
org.hibernate.jdbc.ConnectionManager.openConnection(ConnectionManager.java:420)
        at 
org.hibernate.jdbc.ConnectionManager.getConnection(ConnectionManager.java:144)
        at 
org.hibernate.jdbc.AbstractBatcher.prepareQueryStatement(AbstractBatcher.java:105)
        at 
org.hibernate.loader.Loader.prepareQueryStatement(Loader.java:1561)
        at org.hibernate.loader.Loader.scroll(Loader.java:2217)
        at 
org.hibernate.loader.criteria.CriteriaLoader.scroll(CriteriaLoader.java:89)
        at org.hibernate.impl.SessionImpl.scroll(SessionImpl.java:1497)
        at org.hibernate.impl.CriteriaImpl.scroll(CriteriaImpl.java:297)
        at org.hibernate.impl.CriteriaImpl.scroll(CriteriaImpl.java:291)
        at 
net.sf.plantlore.server.HibernateDBLayer.initialize(HibernateDBLayer.java:206)
        ... 4 more
Caused by: org.postgresql.util.PSQLException: Connection rejected: FATAL: 
password authentication failed for user &quot;jpcbs_pluto&quot;.
        at 
org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:275)
        at 
org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:94)
        at 
org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:65)
        at 
org.postgresql.jdbc2.AbstractJdbc2Connection.&lt;init&gt;(AbstractJdbc2Connection.java:117)
        at 
org.postgresql.jdbc2.Jdbc2Connection.&lt;init&gt;(Jdbc2Connection.java:24)
        at org.postgresql.Driver.connect(Driver.java:235)
        at java.sql.DriverManager.getConnection(DriverManager.java:525)
        at java.sql.DriverManager.getConnection(DriverManager.java:140)
        at 
org.hibernate.connection.DriverManagerConnectionProvider.getConnection(DriverManagerConnectionProvider.java:110)
        at 
org.hibernate.jdbc.ConnectionManager.openConnection(ConnectionManager.java:417)
        ... 13 more
a instsance, kter&#225; to zablokovala hl&#225;s&#237;:
18:47:00,453 ERROR Thread-6 JDBCExceptionReporter:72 - Batch entry 0 
insert into THABITATS (CVERSION, CTERRITORYID, CPHYTOCHORIAID, CQUADRANT, 
CDESCRIPTION, CNEARESTVILLAGEID, CCOUNTRY, CALTITUDE, CLATITUDE, 
CLONGITUDE, CNOTE, CCREATEWHO, CDELETE, CID) values (0, 11, 234, NULL, 
okjl, 44, NULL, NULL, NULL, NULL, khgcf, 2, 0, 31) was aborted.  Call 
getNextException to see the cause.
18:47:00,453 ERROR Thread-6 JDBCExceptionReporter:72 - ERROR: duplicate 
key violates unique constraint &quot;thabitats_pkey&quot;
18:47:00,468 ERROR Thread-6 AbstractFlushingEventListener:300 - Could not 
synchronize database state with session
org.hibernate.exception.ConstraintViolationException: Could not execute 
JDBC batch update
        at 
org.hibernate.exception.SQLStateConverter.convert(SQLStateConverter.java:71)
net.sf.plantlore.common.exception.DBLayerException: 
Exception.CommitTransaction
        at 
net.sf.plantlore.server.HibernateDBLayer.commitTransaction(HibernateDBLayer.java:914)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at 
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
        at 
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:585)
        at 
org.hibernate.exception.JDBCExceptionHelper.convert(JDBCExceptionHelper.java:43)
        at spin.Invocation.evaluate(Invocation.java:175)
        at spin.off.SpinOffEvaluator.evaluate(SpinOffEvaluator.java:120)
        at spin.ProxyFactory.evaluteInvocation(ProxyFactory.java:85)
        at 
spin.JDKProxyFactory$SpinInvocationHandler.invoke(JDKProxyFactory.java:102)
        at $Proxy0.commitTransaction(Unknown Source)
        at 
net.sf.plantlore.client.login.Login$DBLayerProxy.commitTransaction(Login.java:590)
        at 
net.sf.plantlore.client.overview.AddEdit$1.task(AddEdit.java:1232)
        at net.sf.plantlore.common.Task$1.construct(Task.java:200)
        at net.sf.plantlore.common.SwingWorker$2.run(SwingWorker.java:108)
        at java.lang.Thread.run(Thread.java:595)
Caused by: org.hibernate.exception.ConstraintViolationException: Could not 
execute JDBC batch update
        at 
org.hibernate.exception.SQLStateConverter.convert(SQLStateConverter.java:71)
        at 
org.hibernate.exception.JDBCExceptionHelper.convert(JDBCExceptionHelper.java:43)
        at 
org.hibernate.jdbc.AbstractBatcher.executeBatch(AbstractBatcher.java:202)
        at 
org.hibernate.jdbc.AbstractBatcher.prepareStatement(AbstractBatcher.java:91)
        at 
org.hibernate.jdbc.AbstractBatcher.prepareStatement(AbstractBatcher.java:86)
        at 
org.hibernate.jdbc.AbstractBatcher.prepareBatchStatement(AbstractBatcher.java:171)
        at 
org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:2048)
        at 
org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:2427)
        at 
org.hibernate.action.EntityInsertAction.execute(EntityInsertAction.java:51)
        at org.hibernate.engine.ActionQueue.execute(ActionQueue.java:248)
        at 
org.hibernate.engine.ActionQueue.executeActions(ActionQueue.java:232)
        at 
org.hibernate.engine.ActionQueue.executeActions(ActionQueue.java:139)
        at 
org.hibernate.event.def.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:297)
        at 
org.hibernate.event.def.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:27)
        at org.hibernate.impl.SessionImpl.flush(SessionImpl.java:985)
        at 
org.hibernate.impl.SessionImpl.managedFlush(SessionImpl.java:333)
        at 
org.hibernate.transaction.JDBCTransaction.commit(JDBCTransaction.java:106)
        at 
net.sf.plantlore.server.HibernateDBLayer.commitTransaction(HibernateDBLayer.java:902)
        ... 14 more
Caused by: java.sql.BatchUpdateException: Batch entry 0 insert into 
THABITATS (CVERSION, CTERRITORYID, CPHYTOCHORIAID, CQUADRANT, 
CDESCRIPTION, CNEARESTVILLAGEID, CCOUNTRY, CALTITUDE, CLATITUDE, 
CLONGITUDE, CNOTE, CCREATEWHO, CDELETE, CID) values (0, 11, 234, NULL, 
okjl, 44, NULL, NULL, NULL, NULL, khgcf, 2, 0, 31) was aborted.  Call 
getNextException to see the cause.
        at 
org.postgresql.jdbc2.AbstractJdbc2Statement$BatchResultHandler.handleError(AbstractJdbc2Statement.java:2392)
        at 
org.postgresql.core.v3.QueryExecutorImpl$1.handleError(QueryExecutorImpl.java:386)
        at 
org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1257)
        at 
org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:334)
        at 
org.postgresql.jdbc2.AbstractJdbc2Statement.executeBatch(AbstractJdbc2Statement.java:2451)
        at 
org.hibernate.jdbc.AbstractBatcher.executeBatch(AbstractBatcher.java:202)
        at 
org.hibernate.jdbc.AbstractBatcher.prepareStatement(AbstractBatcher.java:91)
        at 
org.hibernate.jdbc.AbstractBatcher.prepareStatement(AbstractBatcher.java:86)
        at 
org.hibernate.jdbc.AbstractBatcher.prepareBatchStatement(AbstractBatcher.java:171)
        at 
org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:2048)
        at 
org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:2427)
        at 
org.hibernate.action.EntityInsertAction.execute(EntityInsertAction.java:51)
        at org.hibernate.engine.ActionQueue.execute(ActionQueue.java:248)
        at 
org.hibernate.engine.ActionQueue.executeActions(ActionQueue.java:232)
        at 
org.hibernate.engine.ActionQueue.executeActions(ActionQueue.java:139)
        at 
org.hibernate.event.def.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:297)
        at 
org.hibernate.event.def.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:27)
        at org.hibernate.impl.SessionImpl.flush(SessionImpl.java:985)
        at 
org.hibernate.impl.SessionImpl.managedFlush(SessionImpl.java:333)
        at 
org.hibernate.transaction.JDBCTransaction.commit(JDBCTransaction.java:106)
        at 
net.sf.plantlore.server.HibernateDBLayer.commitTransaction(HibernateDBLayer.java:902)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at 
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
        at 
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:585)
        at spin.Invocation.evaluate(Invocation.java:175)
        at spin.off.SpinOffEvaluator.evaluate(SpinOffEvaluator.java:120)
        at spin.ProxyFactory.evaluteInvocation(ProxyFactory.java:85)
        at 
spin.JDKProxyFactory$SpinInvocationHandler.invoke(JDKProxyFactory.java:102)
        at $Proxy0.commitTransaction(Unknown Source)
        at 
net.sf.plantlore.client.login.Login$DBLayerProxy.commitTransaction(Login.java:590)
        at 
net.sf.plantlore.client.overview.AddEdit$1.task(AddEdit.java:1232)
        at net.sf.plantlore.common.Task$1.construct(Task.java:200)
        at net.sf.plantlore.common.SwingWorker$2.run(SwingWorker.java:108)
        at java.lang.Thread.run(Thread.java:595)
Caused by: java.sql.BatchUpdateException: Batch entry 0 insert into 
THABITATS (CVERSION, CTERRITORYID, CPHYTOCHORIAID, CQUADRANT, 
CDESCRIPTION, CNEARESTVILLAGEID, CCOUNTRY, CALTITUDE, CLATITUDE, 
CLONGITUDE, CNOTE, CCREATEWHO, CDELETE, CID) values (0, 11, 234, NULL, 
okjl, 44, NULL, NULL, NULL, NULL, khgcf, 2, 0, 31) was aborted.  Call 
getNextException to see the cause.
        at 
org.postgresql.jdbc2.AbstractJdbc2Statement$BatchResultHandler.handleError(AbstractJdbc2Statement.java:2392)
        at 
org.postgresql.core.v3.QueryExecutorImpl$1.handleError(QueryExecutorImpl.java:386)
        at 
org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1257)
        at 
org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:334)
        at 
org.postgresql.jdbc2.AbstractJdbc2Statement.executeBatch(AbstractJdbc2Statement.java:2451)
        at 
org.hibernate.jdbc.BatchingBatcher.doExecuteBatch(BatchingBatcher.java:58)
        at 
org.hibernate.jdbc.AbstractBatcher.executeBatch(AbstractBatcher.java:195)
        ... 29 more
18:47:00,468 FATAL Thread-6 server:913 - JDBC Exception caught while 
commiting database transaction. SQL State: null; Details: Could not 
execute JDBC batch update
18:47:00,484 DEBUG Thread-6 common:163 - ProgressBarManager  received 
message: STOPPING
18:47:00,484 DEBUG AWT-EventQueue-0 common:114 - ProgressBarManager 
STOPPING Task(28410421)
18:47:00,484 DEBUG AWT-EventQueue-0 common:90 - Dispatcher: Task(28410421) 
finished, free to dispatch another.
        at 
org.hibernate.jdbc.BatchingBatcher.doExecuteBatch(BatchingBatcher.java:58)
        at 
org.hibernate.jdbc.AbstractBatcher.executeBatch(AbstractBatcher.java:195)
        ... 29 more
D&#283;l&#225; to i kdy&#382; to spust&#237;m na samostatn&#253;ch kompech

Edituji-li st&#225;vaj&#237;c&#237; z&#225;znam a u n&#225;lezu p&#345;id&#225;m jen revizora, tak mi to 
zahl&#225;s&#237;, &#382;e chci p&#345;idat stejn&#253; z&#225;znam a nic to neud&#283;l&#225;.

Vytvo&#345;&#237;m-li autora a nem&#225;m admin pr&#225;va k datab&#225;zi (m&#367;&#382;u jen p&#345;id&#225;vat 
data), tak ho nem&#367;&#382;u editovat ani smazat.

Jak sma&#382;u lokalitu?
V seznamu taxon&#367; je hnusn&#225; chybka, v sloupci tplants.ctaxon je u taxon&#367; 
kon&#269;&#237;c&#237; na sp. nebo agg. za t&#237;mto mezera. Biocase s t&#237;m m&#225; probl&#233;my a 
plantlore takov&#253;to n&#225;lez nenaimportuje. Jestli&#382;e jsem tuto koncouvou 
mezeru v sloupci tplants.ctaxon odstranil, pak funguje biocase i import v 
plantlore.

Vl&#225;&#271;a
-------------- next part --------------
An HTML attachment was scrubbed...
URL: &lt;<A HREF="https://lists.berlios.de/pipermail/plantlore-dev/attachments/20070103/ed5240bc/attachment.html">https://lists.berlios.de/pipermail/plantlore-dev/attachments/20070103/ed5240bc/attachment.html</A>&gt;
</PRE>

<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="001336.html">[Plantlore-dev] Poznamky k pripominkam a chybam
</A></li>
	<LI>Next message: <A HREF="001338.html">[Plantlore-dev] Pracovni schuzka
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#1337">[ date ]</a>
              <a href="thread.html#1337">[ thread ]</a>
              <a href="subject.html#1337">[ subject ]</a>
              <a href="author.html#1337">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="https://lists.berlios.de/mailman/listinfo/plantlore-dev">More information about the Plantlore-dev
mailing list</a><br>
</body></html>
